### PR TITLE
feat: Add sort_by and cluster_by to expression builder

### PR DESCRIPTION
**Pull request by robert8138**
We already have SORT & CLUSTER expression and SORT_BY & CLUSTER_BY tokens, they are currently used in DDLs but they can also be used in expression builder. This PR adds that support.

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -1155,6 +1155,76 @@ def order_by(self, *expressions, append=True, dialect=None, copy=True, **opts):
             **opts,
         )
 
+    def sort_by(
+        self, *expressions, append=True, dialect=None, parser_opts=None, copy=True
```
Comment (tobymao) [inline sqlglot/expressions.py:1159]: change parse_opts into **opts kwargs, look at the other examples

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -1155,6 +1155,76 @@ def order_by(self, *expressions, append=True, dialect=None, copy=True, **opts):
...
+    ):
+        """
+        Set the CLUSTER BY expression.
```
Comment (tobymao) [inline sqlglot/expressions.py:1162]: sort by

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -1155,6 +1155,76 @@ def order_by(self, *expressions, append=True, dialect=None, copy=True, **opts):
...
+            append=append,
+            copy=copy,
+            prefix="SORT BY",
+            into=Sort,
+            dialect=dialect,
+            parser_opts=parser_opts,
+        )
+
+    def cluster_by(
+        self, *expressions, append=True, dialect=None, parser_opts=None, copy=True
```
Comment (tobymao) [inline sqlglot/expressions.py:1194]: opts kwargs, don't forget to change the docstring as well

**Comment by robert8138** [comment]
Thanks for the quick review. Addressed comments.

**Review by tobymao — APPROVED** [review]

**Pull request merged**